File: typescript/resistor-color-duo/resistor-color-duo.ts
export function decodedValue(resistorColours: string[]) {
  console.log(resistorColours)

  const valueOne = resistorColours[0]
  const valueTwo = resistorColours[1]

  console.log(valueOne)

  const colours = {
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9,
  };
  console.log((colours));
}

File: ruby/resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo

  def self.value(str)
    colours = ["black","brown","red","orange","yellow","green","blue","violet","grey","white"]
  
    p str
    number = []
    number.push(colours.find_index(str[0]))
    number.push(colours.find_index(str[1]))
    p number.join().to_i
  end
end